packages/legend-extension-dsl-data-space/src/stores/query-builder/DataSpaceQueryBuilderState.ts

```diff
@@ -84,11 +82,34 @@ export const resolveUsableDataSpaceClasses = (
   dataSpace: DataSpace,
   mapping: Mapping,
   graphManagerState: GraphManagerState,
+  queryBuilderState?: DataSpaceQueryBuilderState,
 ): Class[] => {
-  const compatibleClasses = getMappingCompatibleClasses(
+  let compatibleClasses = getMappingCompatibleClasses(
     mapping,
     graphManagerState.usableClasses,
   );
+  const mappingModelCoverageAnalysisResult =
+    queryBuilderState?.dataSpaceAnalysisResult?.executionContextsIndex.get(
+      queryBuilderState.executionContext.name,
+    )?.mappingModelCoverageAnalysisResult;
+  if (
```

Comment:
HEADS-UP: This function is not the same as the one in https://github.com/finos/legend-studio/pull/2523